dir.h

```diff
@@ -12,24 +12,24 @@ struct dir_entry {
 	char name[FLEX_ARRAY]; /* more */
 };
 
-#define EXC_FLAG_NODIR 1
-#define EXC_FLAG_ENDSWITH 4
-#define EXC_FLAG_MUSTBEDIR 8
-#define EXC_FLAG_NEGATIVE 16
+#define PATTERN_FLAG_NODIR 1
+#define PATTERN_FLAG_ENDSWITH 4
+#define PATTERN_FLAG_MUSTBEDIR 8
+#define PATTERN_FLAG_NEGATIVE 16
```

Comment:
Git style question (that I didn't see covered in the code guidelines ... and I realize you're not changing it either but thought I'd ask 😄):

I've seen a lot of spots in the code where `1 <<` is used for flag values.  Do you know if there is a preferred style for flags in general?

Reply:
`(1 << x)` makes it clearer that they are independent flags. this is a really old area of code.